src/interfaces/coral_web/src/components/Agents/CreateAgent.tsx

```diff
@@ -54,26 +55,35 @@ export const CreateAgent: React.FC = () => {
 
   const openFilePicker = useOpenGoogleDrivePicker((data) => {
     if (data.docs) {
-      setFields((prev) => ({
-        ...prev,
-        tools_metadata: [
-          ...(prev.tools_metadata?.filter((tool) => tool.tool_name !== TOOL_GOOGLE_DRIVE_ID) ?? []),
-          ...[
-            {
-              tool_name: TOOL_GOOGLE_DRIVE_ID,
-              artifacts: data.docs.map(
-                (doc: any) =>
-                  ({
-                    id: doc.id,
-                    name: doc.name,
-                    type: doc.type,
-                    url: doc.url,
-                  } as GoogleDriveToolArtifact)
-              ),
-            },
+      setFields((prev) => {
+        const updatedArtifacts = [
+          ...(prev.tools_metadata?.find((tool) => tool.tool_name === TOOL_GOOGLE_DRIVE_ID)
```

Comment:
Can we extract this function into a variable? so we can use it later, like:
```tsx
const hasGoogleDriveTool = prev.tools_metadata?.find((tool) => tool.tool_name === TOOL_GOOGLE_DRIVE_ID;
```

Reply:
I don't find very useful to do this abstraction, in this file it is referenced only here

Reply:
the other place where we iterate over tools, is to select all, except google drive tool, so this will also not be very helpful